File: forloop-add-test/forloop_add_test.rs
// rust -O forloop-add-test.rs

use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    let toval: i64 = args[1].parse().unwrap();
    let mut sumval = 0;
    for i in 0..toval {
        for j in 0..toval {
            sumval += i+(j^1);
        }
    }
    println!("Rust: {}", sumval);
}

File: forloop-add-test/src/forloop_add_test.cs
public static class forloop_add_test {
    static void Main(string[] args) {
        long toval = long.Parse(args[0]);
        long i, j, sumval = 0;
        for (i=0; i<toval; i++) {
            for (j=0; j<toval; j++) {
                sumval += i+(j^1);
            }
        }
        System.Console.WriteLine("C# (mono): "+sumval);
    }
}

